src/sys/select.rs

```diff
@@ -85,3 +122,84 @@ pub fn select(nfds: c_int,
 
     Errno::result(res)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use sys::time::{TimeVal, TimeValLike};
+    use unistd::{write, pipe};
+
+    #[test]
+    fn test_fdset() {
```

Comment:
This should probably be broken into a few smaller tests that make it more clear as to what's being tested. Like break up the adding an `Fd` into a test, then test adding & removing, then test clear in a separate test, etc. It makes debugging failures much easier if tests are grouped into a small amount of related tests. This has all the tests in one function.